### ISSUE TITLE
Some floating-point numbers are changed

### ISSUE THREAD

**Issue by ErikSchierboom** [fsprojects/fantomas#785]
Issue created from [fantomas-ui](https://jindraivanek.gitlab.io/fantomas-ui/#?code=LYewJgrgNgpgBAOQsARjATgZwLACg+wAucAHnALxwC0AzAHQBMAjDQGYEzECeFcTdANmbtcROAC9eVfgFYRYgIa96zNkA)

The value of some floating-point numbers will change when formatting the code. It appears to only apply to negative floating-point numbers:

#### Code

```fsharp
module Numbers

let x = -3.213f
let y = 1.621f
let z = -1.5f
let a = 3.213f
```
            

#### Result

```fsharp
module Numbers

let x = -3.213000059f
let y = 1.621f
let z = -1.5f
let a = 3.213f

```
            
#### Options

Fantomas 3.2.0

| Name | Value |
| ---- | ----- |
| `IndentOnTryWith` | `false` |
| `IndentSpaceNum` | `4` |
| `KeepNewlineAfter` | `false` |
| `MaxIfThenElseShortWidth` | `40` |
| `PageWidth` | `120` |
| `ReorderOpenDeclaration` | `false` |
| `SemicolonAtEndOfLine` | `false` |
| `SpaceAfterComma` | `true` |
| `SpaceAfterSemicolon` | `true` |
| `SpaceAroundDelimiter` | `true` |
| `SpaceBeforeArgument` | `true` |
| `SpaceBeforeColon` | `false` |
| `StrictMode` | `false` |

**Comment by ErikSchierboom** [comment]
I've dug into this a bit more, and it looks like this is not Fantomas' doing but FSC: https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYAHmAC8YALQBmAHQAmAIzj4kEAF8gA

**Comment by nojaf** [comment]
Yes, there seems to be a problem in the FCS indeed, however, there is also a Fantomas one here I believe.

Numbers can't be trusted from the AST anyways, so we use trivia to set the correct number, which in this case also didn't happen correctly.

**Comment by nojaf** [comment]
![image](https://user-images.githubusercontent.com/2621499/80911293-b5236400-8d35-11ea-9967-ef1c25a7947f.png)

![image](https://user-images.githubusercontent.com/2621499/80911301-bce30880-8d35-11ea-9f3f-95e057e20c38.png)

We detected the number, the problem is somewhere around the assignment of the trivia to a trivia node.

Given this insight @ErikSchierboom, could I persuade you to tackle this issue yourself?

**Comment by nojaf** [comment]
I believe the problem is located in [here](https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/Trivia.fs#L328).
The minus sign is part of the SynConst range and that is why the match is not found.

Proposed solution:
extend [TokenParser](https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/TokenParser.fs#L413) to have a case of MINUS + NUMBER TOKEN.

Something along the lines:
```fsharp
| minus::head::rest when (minus.TokenInfo.TokenName = "MINUS" && isNumber head) ->
...
```

**Comment by ErikSchierboom** [comment]
Yeah sure! It will be my pleasure :) I'll work on this tomorrow.

**Comment by nojaf** [comment]
Great to hear! I recommend you watch https://www.youtube.com/watch?v=FCrkUqCCzgI&list=PLvw_J2kfZCX3Mf6tEbIPZXbzJOD1VGl4K first to have some understanding of what all this trivia fuzz is about.

If you were to have any questions, please reach out over here or on the F# foundation slack.

**Comment by ErikSchierboom** [comment]
Will do! Thanks for the links.

**Comment by ErikSchierboom** [comment]
@nojaf I've submitted [a PR to fix this](#808). I have one question about how to best implement it, which I've left as a comment in the PR.
